utility/BloomFilter.hpp

```diff
@@ -47,7 +53,30 @@ class BloomFilter {
 
   /**
    * @brief Constructor.
-   * @note The ownership of the bit array lies with the caller.
+   * @note When no bit_array is passed as an argument to the constructor,
+   *       then the bit_array is owned and managed by this class.
+   *
+   * @param random_seed A random_seed that generates unique hash functions.
+   * @param hash_fn_count The number of hash functions used by this bloom filter.
+   * @param bit_array_size_in_bytes Size of the bit array.
+   **/
+  BloomFilter(const std::uint64_t random_seed,
+              const std::size_t hash_fn_count,
+              const std::uint64_t bit_array_size_in_bytes)
+      : random_seed_(random_seed),
+        hash_fn_count_(hash_fn_count),
+        array_size_in_bytes_(bit_array_size_in_bytes),
+        is_bit_array_owner_(true) {
+    array_size_ = bit_array_size_in_bytes * kNumBitsPerByte;
+    bit_array_.reset(new std::uint8_t[bit_array_size_in_bytes]);
```

Comment:
We could move both `array_size_` and `bit_array_` in the init list of the constructor as well.